compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3098,7 +3098,7 @@ private BType checkInvocationParam(BLangInvocation iExpr) {
                 case NAMED_ARGS_EXPR:
                     BVarSymbol varSymbol = params.get(((BLangNamedArgsExpression) expr).name.value);
                     if (!env.enclPkg.packageID.equals(iExpr.symbol.pkgID)
-                            && !Symbols.isFlagOn(varSymbol.flags, Flags.PUBLIC)) {
+                            && (varSymbol != null && !Symbols.isFlagOn(varSymbol.flags, Flags.PUBLIC))) {
```

Comment:
Shouldn't we see why this var symbol is null in the first place?

Reply:
Yeah varSymbol is NULL because it is not available in that function signature.

```ballerina
foo(1, 2, nonExixtent=1000);
foo(int a, int b) {...}
```

This causes compiler to die, which should not be the case. Rather it should be a diagnostic message.

Use case is you often get these scenarios when you migrate code from 0.991.0 to 1.0.0 due to API changes.

Reply:
No need, this happens when the user gives a named arg in function invocation, which is not defined in the function signature. So `params` won't contain that param, hence resulting in a null.